jsco4180Sup/jsco4180.db

```diff
@@ -916,20 +640,23 @@ record(ai, "$(P)COMP:D")
     field(SDIS, "$(P)DISABLE")
 }
 
-record(bo, "$(P)_COMP:SP") 
+record(calc, "$(P)COMP:D:SP") 
 {
-    field(DESC, "Ramp composition component")
-    field(DTYP, "stream")
-    field(ZNAM, "")
-    field(ONAM, "Set")
-    
-    field(OUT, "@jsco4180.proto _setComp($(P)COMP:,TIME:SP,A:SP,B:SP,C:SP) $(PORT)")
+    field(DESC, "Component D % set point")  
+    field(HIHI, "101")
+    field(LOLO, "-1")
     
-    field(SIML, "$(P)SIM")
-    field(SIOL, "$(P)SIM:_COMP:SP")
-    field(SDIS, "$(P)DISABLE")
+    field(INPA, "$(P)COMP:A:SP")
+    field(INPB, "$(P)COMP:B:SP")
+    field(INPC, "$(P)COMP:C:SP")
```

Comment:
Is there a risk that the setpoints here will be uninitialised?

For this type of operation we would usually use the readbacks, and update the readbacks by record redirection at the protocol level after setting any setpoint to ensure they are always up to date.

Reply:
In this instance the record is using the `:SP:RBV` and not the `:RBV`. This is because the device only provides a read back for the component values that have been set as part of a previous program run.